src/symbols/bazel_build_symbol_provider.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as vscode from "vscode";
+import { DocumentSymbolProvider } from "vscode";
+
+import {
+  getBazelWorkspaceFolder,
+  getTargetsForBuildFile,
+  QueryLocation,
+} from "../bazel";
+import { blaze_query } from "../protos";
+
+/** Provids Symbols for targets in Bazel BUILD files. */
+export class StarlarkSymbolProvider implements DocumentSymbolProvider {
```

Comment:
Can you rename this to be `BazelTargetSymbolProvider` (and rename the file to `bazel_target_symbol_provider.ts`? That will more accurately reflect what kind of "symbols" we're providing here.